frontend/pkg/frontend/frontend.go

```diff
@@ -512,6 +515,7 @@ func (f *Frontend) ArmResourceCreateOrUpdate(writer http.ResponseWriter, request
 			arm.WriteInternalServerError(writer)
 			return
 		}
+		tracing.SetClusterAttributes(trace.SpanFromContext(ctx), csCluster)
```

Comment:
Why is `tracing.SetClusterAttributes` being called twice in this function?  Seems like after this "if" block might be a better place, then you would catch it for both creating and updating operations.